Tests/UITests/SlideUpViewSpec.swift

```diff
@@ -40,6 +40,20 @@ class SlideUpViewSpec: QuickSpec {
 
         describe("Slide-up campaign view") {
 
+            context("trigger action with invalid message payload") {
+
+                beforeEach {
+                    launchAppIfNecessary(context: "slide-up-trigger-without-messageBody")
+                    if !iamView.exists {
```

Comment:
Yeah, In this case instead of this condition you should check if iamView is not presented: `expect(iamView.exists).to(beFalse())` and `app.buttons["purchase_successful"].tap()`

Reply:
I don't think, it makes any difference from the current implementation. so it's fine with `if condition`

Reply:
This condition just does not match the expected behaviour. In other contexts it ensures the iamView is present before running any tests cases. In this context you don't want that to happen. So just having `app.buttons["purchase_successful"].tap()` should do

Reply:
ok got it, I understood other contexts `if condition` in a different way. So `expect(iamView.exists).to(beFalse())` and if `!iamView.exists {` is not required, we can directly call `app.buttons["purchase_successful"].tap()`. Will update this.